### ISSUE TITLE
Strange prediction from RAMS

### ISSUE THREAD

**Issue by robmarkcole** [isaaccorley/torchrs#6]
I've passed the first sample from the probav dataset through the RAMS network and get an odd result, is this correct? 

![image](https://user-images.githubusercontent.com/11855322/127016033-6876029a-8c48-47a4-9e68-449d9a7f0176.png)

**Comment by isaaccorley** [comment]
Yes, I haven't finished cleaning up my training script but the grid is caused by the PixelShuffle layers and the input/output being similar is due to the global residual branch. Seems about right.

**Comment by isaaccorley** [comment]
Actually that's not correct, the sr output should be of shape (1, 384, 384). I think it's because you used t=28. They use t=9 in the paper/code. I referenced their tensorflow implementation for some of the code for the temporal reduction block and I noticed it only plays nicely with some values of t.

**Comment by robmarkcole** [comment]
I get the same result for t=9

**Comment by isaaccorley** [comment]
Interesting, I thought my test checked for this. I'll be able to look at it later today. Thanks for trying it out.

**Comment by isaaccorley** [comment]
Ok so I'm unable to replicate the output of `(1, 2, 384, 384)` for `t=9`, however this is true for `t=28`. In the [paper](https://www.mdpi.com/2072-4292/12/14/2207) they set `t=9` and select 9 images randomly from the set of LR images. The `t` and `kernel size` values are used to compute how many temporal reduction blocks are used with the equation ` np.floor((channels-1)/(kernel_size-1)-1).astype(int)` [here from their tensorflow implementation](https://github.com/EscVM/RAMS/blob/3244abbb41653e3f7f9dd3bff318586132a5ab48/utils/network.py#L132).

If you plug `t=9` and `kernel_size=3` into that equation (minus the np.floor) you get `3 blocks`, however if you use `t=28` you get `12.5 blocks` which I believe is causing the output dims to be different.

I did some quick maths and looks like it works for `t = [9, 11, 13, 15, 17, ...], where t>=9` I added some assert statements in the model to ensure this with a warning and some suggested values for `t`. I also updated the tests. Let me know if this works for you.

On a side note, I'll be adding some more models soon which aren't constrained by t>=9.

**Comment by isaaccorley** [comment]
See #7

**Comment by robmarkcole** [comment]
Hmm I get this issue for every `t` I try. I cant rule out that I am doing something dumb in the post processing or visualisation. Please see https://github.com/robmarkcole/scratchpad/blob/main/rams-quickstart.ipynb

**Comment by isaaccorley** [comment]
Seems like it's working fine. In your notebook when you print sr.shape you got (1,1,384,384) which is the correct shape (b,c,h,w). It would only be wrong if the channel dim was not equal to your input channels.

**Comment by robmarkcole** [comment]
I do sometimes get `(1,2,384,384)` even for single channel..

**Comment by isaaccorley** [comment]
This expected though, no? RAMS uses `nn.PixelShuffle` to increase the resolution which creates the pattern you're seeing. It's also not a trained model so I would expect it to not have learned how to upsample properly.

<img src="https://user-images.githubusercontent.com/22203655/127407880-4ac97578-b286-4d19-92bc-7276bf31642a.png" alt="drawing" width="400"/>

If you simply upsample using a single `PixelShuffle` layer with `scale_factor=3`:

```python
import einops
import torch.nn as nn
from torchrs.datasets import PROBAV

dataset = torchrs.datasets.PROBAV(split="train", band="RED")
sample = dataset[0]
lr = sample["lr"][:9]
lr = einops.rearrange(lr, "t c h w -> (t c) h w")
sr = nn.PixelShuffle(3)(lr)
hr = sample["hr"]
```

And then plot `sr` and `hr` you get:

![image](https://user-images.githubusercontent.com/22203655/127411298-3c40356f-1972-4435-9f21-5546e55ed60a.png)

And on closer inspection of the top left corner of the `sr` image.

![image](https://user-images.githubusercontent.com/22203655/127411605-60533c61-fc95-4c34-b497-abfc6d4b70fc.png)

Of course, `RAMS` has multiple `PixelShuffle` ops and they are summed together so I would expect the checkerboard artifacts to be exacerbated like you are seeing.

**Comment by robmarkcole** [comment]
`It's also not a trained model` how did I overlook this.. Thanks!
